app/src/main/java/com/example/bigowlapp/utils/AlarmBroadcastReceiverManager.java

```diff
@@ -0,0 +1,120 @@
+        this.context = context;
+        this.scheduleRepository = scheduleRepository;
+    }
+    // TODO: --END-- REMOVE BEFORE MERGING PULL REQUEST. THIS IS ONLY FOR TESTING PURPOSES.
+
+    /**
+     * Sets the alarm(s) for the BroadcastReceiver given the schedules that the user has
+     *
+     * @param userID The Id of the user
+     */
+    public void setAlarms(String userID) {
+        int requestCode = 0;
+        AlarmManager alarmManager = (AlarmManager) context.getSystemService(Context.ALARM_SERVICE);
+        getSchedulesForUser(userID)
+                .addOnSuccessListener(scheduleList -> {
+                    for (Schedule schedule : scheduleList) {
+                        Intent intent = new Intent(context, AlarmBroadcastReceiver.class);
+                        intent.putExtra(EXTRA_UID, schedule.getUid());
+                        intent.putExtra(EXTRA_LATITUDE, schedule.getLocation().getLatitude());
+                        intent.putExtra(EXTRA_LONGITUDE, schedule.getLocation().getLongitude());
+                        PendingIntent pendingIntent = PendingIntent
+                                .getBroadcast(context, requestCode, intent, PendingIntent.FLAG_UPDATE_CURRENT);
+                        alarmManager.set(AlarmManager.RTC_WAKEUP,
+                                schedule.getStartTime().toDate().getTime(), pendingIntent);
+                    }
+                }).addOnFailureListener(error -> Toast.makeText(context, error.getMessage(), Toast.LENGTH_SHORT).show());
+    }
+
+    /**
+     * Obtains the list of schedules for the user, and filters in schedules that have been accepted
+     *
+     * @param userID The Id of the user
+     * @return A Task that contains a list of schedule for the user that hasn't been attended
+     */
+    private Task<List<Schedule>> getSchedulesForUser(String userID) {
+
+        // TODO: --START-- REMOVE BEFORE MERGING PULL REQUEST. THIS IS ONLY FOR TESTING PURPOSES.
+        if (userID.equals("testing")) {
+            List<Schedule> acceptedScheduleList = new ArrayList<>();
+
+            Schedule schedule = new Schedule();
+            schedule.setUid("Hello there");
+            Calendar calendar = Calendar.getInstance();
+            calendar.setTime(new Date());
+            calendar.add(Calendar.SECOND, 30);
+            Timestamp timestamp = new Timestamp(calendar.getTime());
+            schedule.setStartTime(timestamp);
+            GeoPoint geoPoint = new GeoPoint(10, -82);
+            schedule.setLocation(geoPoint);
+
+            acceptedScheduleList.add(schedule);
+
+            return Tasks.forResult(acceptedScheduleList);
+        }
+        // TODO: --END-- REMOVE BEFORE MERGING PULL REQUEST. THIS IS ONLY FOR TESTING PURPOSES.
+
+        return scheduleRepository.getTaskListSchedulesForUser(userID)
+                .continueWithTask(task -> {
+                    if (task.isSuccessful()) {
```

Comment:
can use OnSuccessTask to reduce complexity (as the if statement would be unnecessary), but not necessary change.

Reply:
Problem is that onSuccessTask doesn't return an exception, so if there's a problem/error, there won't be a way to handle it.

I'm not sure if this is a function that doesn't require exception handling.

Reply:
@ShawnGuerra-Bautista It actually does, you just don't need to specifically define it.

As for the disadvantage of OnSuccessTask, if you need to do something other than throw an error on failure, you can't. Although looking at the code right now, the else statement only throws the error, hence my suggestion of OnSuccessTask.

Behavior should be exactly the same.

Reply:
https://developers.google.com/android/reference/com/google/android/gms/tasks/Task#onSuccessTask(com.google.android.gms.tasks.SuccessContinuation<TResult, TContinuationResult>)

Alright, I re-read it, and I believe you are right. So, I'll do the changes.